pkg/agent/interfacestore/interface_cache_test.go

```diff
@@ -65,28 +67,75 @@ func testContainerInterface(t *testing.T) {
 	assert.True(t, exists)
 	_, exists = store.GetInterfaceByIP(podIP.String())
 	assert.True(t, exists)
+	_, exists = store.GetInterfaceByIP(podIPv6.String())
+	assert.True(t, exists)
 	_, exists = store.GetInterfaceByOFPort(uint32(containerInterface.OVSPortConfig.OFPort))
 	assert.True(t, exists)
 	ifaces := store.GetContainerInterfacesByPod(containerInterface.PodName, containerInterface.PodNamespace)
 	assert.Equal(t, 1, len(ifaces))
-	assert.True(t, reflect.DeepEqual(ifaces[0], containerInterface))
+	assert.Equal(t, containerInterface, ifaces[0])
 	ifaceNames := store.GetInterfaceKeysByType(ContainerInterface)
 	assert.Equal(t, 1, len(ifaceNames))
 	assert.Equal(t, containerInterfaceKey, ifaceNames[0])
 	assert.Equal(t, 1, store.GetContainerInterfaceNum())
+
 	store.DeleteInterface(containerInterface)
 	assert.Equal(t, 0, store.GetContainerInterfaceNum())
 	_, exists = store.GetContainerInterface(containerInterface.ContainerID)
 	assert.False(t, exists)
 	_, exists = store.GetInterfaceByIP(containerInterface.IPs[0].String())
 	assert.False(t, exists)
+	_, exists = store.GetInterfaceByIP(containerInterface.IPs[1].String())
+	assert.False(t, exists)
+
 	containerInterface.IPs = nil
 	store.AddInterface(containerInterface)
 	assert.Equal(t, 1, store.GetContainerInterfaceNum())
 	_, exists = store.GetInterfaceByIP(podIP.String())
 	assert.False(t, exists)
 }
 
+func testSecondaryInterface(t *testing.T) {
+	store := NewInterfaceStore()
+	// Seondary interface without an IP.
+	containerInterface1 := NewContainerInterface("c0-eth1", "c0", "p0", "ns0", "eth1", podMAC, nil, 2)
+	containerInterface2 := NewContainerInterface("c0-eth2", "c0", "p0", "ns0", "eth2", podMAC, []net.IP{podIP}, 0)
+	store.Initialize([]*InterfaceConfig{containerInterface1, containerInterface2})
+	assert.Equal(t, 2, store.Len())
+
+	for _, containerInterface := range []*InterfaceConfig{containerInterface1, containerInterface2} {
+		interfaceKey := util.GenerateContainerInterfaceKey(containerInterface.ContainerID, containerInterface.IFDev)
+		storedIface, exists := store.GetInterface(interfaceKey)
+		assert.True(t, exists)
```

Comment:
feels like this would be better as a `require.True`

Reply:
Done